platforms/native/platform-native/src/main/java/org/gradle/nativeplatform/internal/modulemap/GenerateModuleMapFile.java

```diff
@@ -16,23 +16,23 @@
 
 package org.gradle.nativeplatform.internal.modulemap;
 
-import com.google.common.collect.Lists;
 import com.google.common.io.Files;
 import org.apache.commons.io.FileUtils;
 import org.gradle.api.UncheckedIOException;
 
 import java.io.File;
 import java.io.IOException;
+import java.util.ArrayList;
 import java.util.List;
 
 import static org.gradle.util.internal.CollectionUtils.collect;
 import static org.gradle.util.internal.CollectionUtils.filter;
 
 public class GenerateModuleMapFile {
     public static void generateFile(File moduleMapFile, String moduleName, List<String> publicHeaderDirs) {
-        List<String> lines = Lists.newArrayList(
-            "module " + moduleName + " {"
-        );
+        String firstLine = "module " + moduleName + " {";
+        List<String> lines = new ArrayList<>();
```

Comment:
💅 Could pre-compute the size needed here, too.

Reply:
I didn't feel this was necessary. This isn't really performance sensitive at all and I didn't want to add the maintenance burden

Reply:
If it were simpler (didnt have a dynamic size based on filtering) I'd say this is more worth it